File: code/csharp/traning/Variables.cs
public class Variables
{
    public static void Run()
    {
        // Explicit type declaration
        string name = "John";
        int age = 30;
        double salary = 75000.50;
        bool isEmployed = true;
        char grade = 'A';

        // Type inference with 'var' (compiler determines type)
        var city = "New York";      // Inferred as string
        var count = 100;            // Inferred as int
        var price = 19.99;          // Inferred as double

        // Constants (immutable)
        const double PI = 3.14159;
        const string COMPANY = "TechCorp";

        // Nullable types (can hold null)
        int? nullableAge = null;
        string? nullableName = null;

        // Print all variables
        Console.WriteLine($"Name: {name}");
        Console.WriteLine($"Age: {age}");
        Console.WriteLine($"Salary: {salary}");
        Console.WriteLine($"Employed: {isEmployed}");
        Console.WriteLine($"Grade: {grade}");
        Console.WriteLine($"City: {city}");
        Console.WriteLine($"PI: {PI}");
        Console.WriteLine($"Nullable Age: {nullableAge ?? 0} \n");
    }
}

File: code/py/variables.py
name = "John"
age = 30
salary = 75000.50
is_employed = True
grade = 'A'

PI = 3.14159
COMPANY = "TechCorp"

name_typed: str = "Jane"
age_typed: int = 25
salary_typed: float = 65000.00
is_active: bool = False

x,y,z = 1,2,3
a =b =c = 0 # all equal to 0

nullable_value = None

dynamic_var = 10
print(f"dynamic_var is {type(dynamic_var)}: {dynamic_var}")

dynamic_var = "Hello"
print(f"dynamic_var is {type(dynamic_var)}: {dynamic_var}")

dynamic_var = [1,2,3]
print(f"dynamic_var is {type(dynamic_var)}: {dynamic_var}")


# Print all variables
print(f"Name: {name}")
print(f"Age: {age}")
print(f"Salary: {salary}")
print(f"Employed: {is_employed}")
print(f"PI: {PI}")
print(f"x, y, z: {x}, {y}, {z}")
print(f"Nullable: {nullable_value}")

# Check types
print(f"\nType checking:")
print(f"name type: {type(name)}")
print(f"age type: {type(age)}")
print(f"salary type: {type(salary)}")
print(f"is_employed type: {type(is_employed)}")

